app/models/get_an_identity/user.rb

```diff
@@ -0,0 +1,54 @@
+module GetAnIdentity
+  class User
+    class InvalidTokenError < StandardError; end
+    class NotFoundError < StandardError; end
+
+    def self.find(id)
+      new(id:)
+    end
+
+    attr_reader :id,
+                :uid,
+                :email,
+                :full_name,
+                :date_of_birth,
+                :trn,
+                :mobile_number,
+                :merged_user_ids,
+                :raw
+
+    def initialize(id:)
+      @id = id
+      self.load
+    end
+
+  private
+
+    def request_user
+      url = "#{ENV.fetch('TRA_OIDC_DOMAIN')}/api/v1/users/#{id}"
+      headers = { "Authorization" => "Bearer #{GetAnIdentity::AccessToken.new}" }
+
+      HTTParty.get(url, headers:).tap do |response|
```

Comment:
I was going to use JsonApiClient like the ECF integration but I had to start building a parser since GAI doesn't use the `{ "data": { "type"/"attributes"/"id": .. }` formatting for their responses so I went with just hitting the API with HTTParty and parsing manually in load like this since overall it led to code that was easier to understand.

Reply:
I think going forward if this was expanded at all we would have a base class for all the configuration related stuff plus use a decorator/presenter/serialiser for the object maybe?

Reply:
So you would have a very small command class like `user_details` that inherits from the base class with all configuration and then the command class which basically describes what the api call is doing would have serialised decorator for how the data should be presented?

Reply:
Yeah that would make sense when things expand. I'm hopeful that with the webhook integration we won't need to do any more integration with APIs.